ui/pages/routes/routes.component.js

```diff
@@ -238,15 +246,37 @@ export default class Routes extends Component {
   ///: END:ONLY_INCLUDE_IF
 
   componentDidUpdate(prevProps) {
-    const { theme, account } = this.props;
-
+    const {
+      theme,
+      account,
+      networkToAutomaticallySwitchTo,
+      activeTabOrigin,
+      unapprovedTransactions,
+      isUnlocked,
+    } = this.props;
     if (theme !== prevProps.theme) {
       this.setTheme();
     }
 
     if (prevProps.account?.address !== account?.address) {
       this.setState({ hideConnectAccountToast: false });
     }
+
+    // Automatically switch the network if the user
+    // no longer has unapprovedTransactions and they
+    // should be on a different network for the
+    // currently active tab's dapp
+    if (
```

Comment:
Nit: Maybe worth considering moving this logic into a higher-order-component, or a look in a HOC? e.g. wrap everything here in an `<AutomaticallySwitchNetwork>...</AutomaticallySwitchNetwork>`. This functionality adds a lot to the Routes props, splitting it out would seem simpler.